src/ast.mli

```diff
@@ -1,83 +1,91 @@
+type program = None
+
 type bin_op =
-    | Add    | Minus | Multiply | Divide
-    | Modulo | Equal | Neq      | Lt
-    | Leq    | Gt    | Geq      | And
-    | Or
+    Add | Minus | Multiply | Divide
+  | Modulo | Equal | Neq | Lt
+  | Leq | Gt | Geq | And
+  | Or
 
 type u_op = Not | Neg
 
 type p_type = Int_t | Bool_t | Double_t | Char_t | Unit_t
 
 type cont_type = String_t | List_t | Set_t | Map_t | Tuple_t
 
-type opt_type = Maybe_t of types | Some_t of types | None_t
+type actor_op =
+    Actor_send
+  | Actor_broadcast
+  | Actor_receive
 
-type message_type = Message_t
+type types =
+    Primitive of p_type
+  | Container of cont_type
+  | None_type
 
-type field = Field of types * string
+type opt_type = Maybe_t | Some_t | None_t
 
-type func = {
-    func_name : string;
-    function_return_t : types;
-    function_formals : formal list;
-    function_body : stmt list;
+type optional = {
+    type_: opt_type;
+    param: types;
 }
 
-type actor = {
-    actor_fields : field list;
-    actor_name : string;
-    acator_receive: receive;
-}
+type formal = string * types
+
+type field = Field of types * string
 
 type actor_type = Actor_t of string
 
 type pool_type = Pool_t of actor_type list
 
-type types =
-    | Primitive of p_type
-    | Container of cont_type
-    | Optional of opt_type
-    | Actor_t of actor_op
-
-type bind_type = types * string
-
 type bit_op =
-    | Big_RShift
-    | Bit_LShift
-    | Bit_And
-    | Bit_Or
-    | Bit_Xor
-    | Bit_Not
+    Big_RShift
+  | Bit_LShift
+  | Bit_And
+  | Bit_Or
+  | Bit_Xor
+  | Bit_Not
+
+type message_type = {
```

Comment:
btw, I had to chain these three types b/c of a cycle:
`message_type -> stmt -> expr -> message_type`